specifyweb/frontend/js_src/lib/components/Formatters/formatters.ts

```diff
@@ -158,11 +161,18 @@ async function formatField(
 
   let formatted: string | undefined = undefined;
   const hasPermission = hasPathPermission(fields ?? [], 'read');
-  if (hasPermission) {
-    const data = await fetchDistantRelated(parentResource, fields);
+  const data = await fetchDistantRelated(parentResource, fields);
+
+  function handleData() {
     if (data === undefined) return { formatted: '' };
     const { resource, field } = data;
     if (field === undefined || resource === undefined) return { formatted: '' };
+    return { resource, field };
+  }
```

Comment:
if this code is called in both branches, without any arguments, then why does it even need to be a function - could just be inlined?

moreover, both branches include this:
```
    const { resource, field } = handleData() ?? {};
    if (!resource || !field) return { formatted: '' };
```

why is that code inside of branches, when it can be put before the branches (given that it's common to both branches)